ruby/ext/google/protobuf_c/defs.c

```diff
@@ -226,6 +226,25 @@ static Descriptor* ruby_to_Descriptor(VALUE val) {
   return ret;
 }
 
+// Decode and return a frozen instance of a Descriptor Option for the given pool
+static VALUE decode_options(const char *option_type, int size, const char *bytes, VALUE descriptor_pool) {
+  static const char * prefix = "google.protobuf.";
+  char fullname[strlen(prefix) + strlen(option_type) + 1];
```

Comment:
We generally try to avoid VLAs. You could use `VALUE fullname = rb_sprintf("%s%s", prefix, option_type)`.

Reply:
Rewritten using `snprintf` - the string in question is just a C string, so avoiding using `rb_*` here.